functions/common/geode-common/src/main/java/org/springframework/cloud/fn/common/geode/GeodeClientCacheConfiguration.java

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * @author David Turanski
+ * @author Christian Tzolov
+ */
+@EnableConfigurationProperties({ GeodeClientCacheProperties.class, GeodeSecurityProperties.class,
+		GeodeSslProperties.class, GeodePoolProperties.class })
+@Import(InetSocketAddressConverterConfiguration.class)
+public class GeodeClientCacheConfiguration {
+
+	private static final String SECURITY_CLIENT = "security-client-auth-init";
+
+	private static final String SECURITY_USERNAME = "security-username";
+
+	private static final String SECURITY_PASSWORD = "security-password";
+
+	@Bean
+	public ClientCache clientCache(GeodeClientCacheProperties clientCacheProperties,
+			GeodeSecurityProperties securityProperties, GeodeSslProperties sslProperties,
+			GeodePoolProperties geodePoolProperties) {
+
+		Properties properties = new Properties();
+		PropertiesBuilder pb = new PropertiesBuilder();
+		if (StringUtils.hasText(securityProperties.getUsername())
+				&& StringUtils.hasText(securityProperties.getPassword())) {
+
+			properties
+					.setProperty(SECURITY_CLIENT,
+							GeodeSecurityProperties.UserAuthInitialize.class.getName() + ".create");
+			properties.setProperty(SECURITY_USERNAME, securityProperties.getUsername());
+			properties.setProperty(SECURITY_PASSWORD, securityProperties.getPassword());
+
+		}
+
+		if (sslProperties.isSslEnabled()) {
+			pb.add(properties);
+			pb.add(this.toGeodeSslProperties(sslProperties));
+		}
+
+		ClientCacheFactory clientCacheFactory = new ClientCacheFactory(pb.build());
+
+		if (clientCacheProperties.isPdxReadSerialized()) {
+			clientCacheFactory.setPdxSerializer(new ReflectionBasedAutoSerializer(".*"));
+			clientCacheFactory.setPdxReadSerialized(true);
+		}
+
+		if (geodePoolProperties.getConnectType().equals(GeodePoolProperties.ConnectType.locator)) {
+			for (InetSocketAddress address : geodePoolProperties.getHostAddresses()) {
+				clientCacheFactory.addPoolLocator(address.getHostName(), address.getPort());
+			}
+		}
+		else {
+			for (InetSocketAddress address : geodePoolProperties.getHostAddresses()) {
+				clientCacheFactory.addPoolServer(address.getHostName(), address.getPort());
+			}
+		}
+
+		clientCacheFactory.setPoolSubscriptionEnabled(geodePoolProperties.isSubscriptionEnabled());
+		ClientCache clientCache = clientCacheFactory.create();
+		clientCache.readyForEvents();
+		return clientCache;
+	}
+
+	/**
+	 * Converts the App Starter properties into Geode native SSL properties.
+	 * @param sslProperties App starter properties.
+	 * @return Returns the geode native SSL properties.
+	 */
+	private Properties toGeodeSslProperties(GeodeSslProperties sslProperties) {
+
+		PropertiesBuilder pb = new PropertiesBuilder();
+
+		// locator - SSL communication with and between locators
+		// server - SSL communication between clients and servers
+		pb.setProperty("ssl-enabled-components", "server,locator");
+
+		pb.setProperty("ssl-keystore", this.resolveRemoteStore(sslProperties.getKeystoreUri(),
```

Comment:
The `PropertiesBuilder` is err, builder: you can just chain its methods.

Reply:
Is it worth changing this? I just copy-pasted what was there.